tei/src/protocol.rs

```diff
@@ -235,12 +236,23 @@ impl fmt::Display for Output {
             Self::Ok => write!(f, "teiok"),
             Self::ReadyOk => write!(f, "readyok"),
             Self::BestMove(the_move) => write!(f, "bestmove {the_move}"),
-            Self::Info { time, nodes, score } => {
+            Self::Info {
+                time,
+                nodes,
+                score,
+                pv,
+            } => {
                 let centipawns = (f32::from(*score) * 100.0) as i32;
                 write!(
                     f,
-                    "info time {} nodes {nodes} score {centipawns}",
-                    time.as_millis()
+                    "info time {} nodes {nodes} nps {} score cp {centipawns} pv {}",
+                    time.as_millis(),
+                    1000 * nodes / time.as_millis() as usize,
+                    pv.iter()
+                        .map(ToString::to_string)
+                        .collect::<Vec<_>>()
+                        .as_slice()
+                        .join(" "),
```

Comment:
This can be written more efficiently with a `fold` to avoid allocating a `String` for every move as well as allocating the `Vec`.

Reply:
I fixed this by just `write!()`ing each element, it's cleaner and should be just as fast with line buffering.

I also noticed that I was printing `score cp` together with `score mate`, I fixed that to only print `mate` when applicable.